### PR TITLE
Update CONTRIBUTING.md

### LINKED ISSUE #1662: Update wiki and CONTRIBUTING to account for docs versions

### PR DESCRIPTION
**Pull request by jseldess**
- Clarify front-matter fields
- Add details about docs versioning
- Add details about "copy to clipboard" feature

Fixes #1662

### CODE REVIEW

**Comment by jseldess** [comment]
@benesch, @sploiselle, PTAL at these updates to `CONTRIBUTING.md`.

Separately, I'll work on updating docs for regenerating sql diagrams now that we have versioning in place.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -150,18 +157,31 @@ The syntax for including content is `{% include <filepath> %}`, for example, `{%
...
+For each documented version of CockroachDB, a JSON file in the `_includes` directory controls which pages appear in the docs sidebar. For example, the sidebar for CockroachDB v1.0 is defined by [`_includes/sidebar-data-v1.0.json`](_includes/sidebar_data-v1.0.json).
+
+If you're adding a page that you think should appear in the sidebar, please mention this in your pull request.
+
+For each page in the sidebar, there are two possible fields:
+
+Field | Description
+------|------------
+`title` | The page title, as displayed in the sidebar. This can be different from the actual page title.
+`urls` | A list of URLs, each formatted as `/${VERSION}/<page-name>.html`, e.g., `/${VERSION}/learn-cockroachdb-sql.html`. The first URL is the page to link to. The subsequent URLs are pages that should highlight this title in the sidebar.
```
Comment (benesch) [inline CONTRIBUTING.md]: Is it worth mentioning `items` here too?

Comment (sploiselle) [inline CONTRIBUTING.md]: What is `items`, @benesch?

**Comment by sploiselle** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1669#-:-KoNB9vDHvrDCmkCrkfG:bnfp4nl)*

File: CONTRIBUTING.md (hunk 1 above)
Comment (jseldess) [inline CONTRIBUTING.md]: @benesch, I've rewritten this section. Please take another look.

Comment (benesch) [inline CONTRIBUTING.md]: Wow, looks great!

**Pull request merged**